x/mongo/driver/operation/count.go

```diff
@@ -102,16 +127,63 @@ func (c *Count) Execute(ctx context.Context) error {
 		ServerAPI:         c.serverAPI,
 	}.Execute(ctx, nil)
 
+	// Swallow error if NamespaceNotFound(26) is returned from aggregate on non-existent namespace
+	if err != nil {
+		dErr, ok := err.(driver.Error)
+		if ok && dErr.Code == 26 {
+			err = nil
+		}
+	}
+	return err
 }
 
 func (c *Count) command(dst []byte, desc description.SelectedServer) ([]byte, error) {
-	dst = bsoncore.AppendStringElement(dst, "count", c.collection)
+	// If wire version < 12 (4.9.0), use count command
+	if desc.WireVersion.Max < 12 {
+		dst = bsoncore.AppendStringElement(dst, "count", c.collection)
+		if c.maxTimeMS != nil {
+			dst = bsoncore.AppendInt64Element(dst, "maxTimeMS", *c.maxTimeMS)
+		}
+		if c.query != nil {
+			dst = bsoncore.AppendDocumentElement(dst, "query", c.query)
+		}
+		return dst, nil
+	}
+	// If wire version >= 12 (4.9.0), use aggregate with $collStats
+	header := bsoncore.Value{Type: bsontype.String, Data: bsoncore.AppendString(nil, c.collection)}
+	if c.collection == "" {
+		header = bsoncore.Value{Type: bsontype.Int32, Data: []byte{0x01, 0x00, 0x00, 0x00}}
+	}
+
+	dst = bsoncore.AppendValueElement(dst, "aggregate", header)
+	cursorIdx, cursorDoc := bsoncore.AppendDocumentStart(nil)
 	if c.maxTimeMS != nil {
 		dst = bsoncore.AppendInt64Element(dst, "maxTimeMS", *c.maxTimeMS)
 	}
 	if c.query != nil {
```

Comment:
The `query` field exists because it's supported by the `count` command but is not supported by `aggregate`. I'd recommend erroring if it's set rather than appending it. This shouldn't have any effect for users of the high-level API because the `EstimatedDocumentCount` helper doesn't support filters anyway.